hs/src/Reach/Eval/Core.hs

```diff
@@ -3721,6 +3723,30 @@ evalPrim p sargs =
         Just (SLV_Bytes _ "UpdateApplication") -> return $ RA_UpdateApplication
         Just (SLV_Bytes _ "DeleteApplication") -> return $ RA_DeleteApplication
         Just _ -> expect_ $ Err_Remote_ALGO_extra $ [ "illegal value for onCompletion" ]
+      -- nonNetRecv default depends on call to bill(). bill() might be called later, so we generate
+      -- the default when the remote fn is called (in later branch, where RemoteFunMode = Nothing)
```

Comment:
I think this should not work that way and instead it should be `Nothing` until specified. I think that you can't call `bill` after `.ALGO` and when you do, it checks that what you're calling it with is consistent with `bill`